tests/test_dataset.py

```diff
@@ -442,3 +450,84 @@ def test_dataset_cache_loading(audb_cache, tmpdir, repository, db, request):
         # to compare it.
         assert str(dataset.header) == str(header)
         assert dataset.repository_object == repository
+
+
+class TestDatasetLoadTables:
+    r"""Test load_tables argument of audbcards.Dataset."""
+
+    @pytest.fixture(autouse=True)
```

Comment:
This very likely works but might be confusing some readers as it mixes two kinds of idioms:

-`def setUp`  (the camelcased version of what you are doing) is a protected method name in class based test implementation from the unittest module.  
- `@pytest.fixture(autouse=True)` is the equivalent from the pytest world.     

Normally people use the fixture outside of the class to flag the dependency injection magic that pytest does. 
Probably to camelcase the method name and to remove the fixture decorator should also work and  read more conventional.  Afaics the setup is done only once for the entire test class. So if you want to be fancy you could   call it `setUpClass` . But then you would have to use the `@classmethod` decorator.

Reply:
> Probably to camelcase the method name and to remove the fixture decorator should also work and read more conventional.

No, `setUp` as well as `setUpClass` are not executed as we are using `unittest`.  
I also see no problem with using the name `setup` as this describes well what it been done here, and can also be used in other classes as well.

I adjusted the code, and changed the `setup()` fixture to a class method.

> Normally people use the fixture outside of the class to flag the dependency injection magic that pytest does.

But isn't the advantage of using the fixture inside the class, that it shows that the fixture is supposed to be used only inside the class. It is also not available outside of the class. The big advantage is we can use `autouse=True` and it is automatically executed before running all other `test_` methods of the class. If it would be defined outside of the class, it would be used automatically in all other test functions, which we don't want.

Reply:
BTW, by using a fixture one would also not need to implement an extra teardown metod, it would just be:

```python
    @classmethod
    @pytest.fixture(autouse=True)
    def setup(cls, ...):
        # prepare some data/variables
        yield
        # add tear down code here
```

Reply:
> I am unsure whether you are using unittest at all as you assume.

As long as `unittest` is not imported, it is not used. In `pytest` you can [directly write classes](https://docs.pytest.org/en/7.1.x/getting-started.html#group-multiple-tests-in-a-class), and don't need to inherit from `unittest.TestCase`.

> So my understanding is that you are using a test class that uses pytest elements.

Exactly.

>  but you are using the `cache_root` argument that I overlooked.

That is correct:
```python
    @classmethod
    @pytest.fixture(autouse=True)
    def setup(cls, cache, medium_db):
        r"""Provide test class with cache and database.

        Args:
            cache: cache fixture
            medium_db: medium_db fixture

        """
        cls.name = medium_db.name
        cls.version = pytest.VERSION
        cls.cache_root = cache
```

But `cache` is another fixture, defined in `tests/conftest.py`, which uses the `tmpdir` fixture, which is automatically destroyed during.

Reply:
I know. My point was that calling a method `setup` and using a `classmethod` might be confusing for developers used to the `unittest.Testcase` idiom as it triggers expectataions towards unittest. 

So I am absolutely fine with the fixture, but  I would rather call this `prepare`  or so to avoid confusion.

Reply:
I think this is a good opportunity to choose a name, that we can reuse in other test classes.

In general, `setup` seems to fit quite well. Instead of confusion, I would say it provides a clue what the method is doing when you are coming from the `unittest` world. But I think `prepare` should also be obvious and I renamed the method to `prepare`.

Reply:
> Semantically `setup` is probably better.

Hehe, we have to decide, `setup` or `prepare`. I also think `setup` is slightly better, but you worried that it might be confusing. So should we go with `setup` or `prepare` here, and in the future?